launcher/MangoHud.cpp

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  PolyMC - Minecraft Launcher
+ *  Copyright (C) 2022 Sefa Eyeoglu <contact@scrumplex.net>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * This file incorporates work covered by the following copyright and
+ * permission notice:
+ *
+ *      Copyright 2013-2021 MultiMC Contributors
+ *
+ *      Authors: Orochimarufan <orochimarufan.x3@gmail.com>
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License");
+ *      you may not use this file except in compliance with the License.
+ *      You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *      Unless required by applicable law or agreed to in writing, software
+ *      distributed under the License is distributed on an "AS IS" BASIS,
+ *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *      See the License for the specific language governing permissions and
+ *      limitations under the License.
+ */
+
+#include "MangoHud.h"
+#include "FileSystem.h"
+#include "Json.h"
+
+namespace MangoHud {
+
+QString getLibraryString()
+{
+    QStringList vkLayerList;
+    QString vkLayerPath = qEnvironmentVariable("VK_LAYER_PATH");
+    if (!vkLayerPath.isEmpty()) {
+        vkLayerList << vkLayerPath;
+    } else {
+        vkLayerList << "/usr/local/etc/vulkan/implicit_layer.d"
+                    << "/usr/local/share/vulkan/implicit_layer.d"
+                    << "/etc/vulkan/implicit_layer.d"
+                    << "/usr/share/vulkan/implicit_layer.d";
+
+        QString data_home = qEnvironmentVariable("XDG_DATA_HOME");
+        if (data_home.isEmpty())
+            data_home = qEnvironmentVariable("HOME");
+
+        if (!data_home.isEmpty())
+            vkLayerList << FS::PathCombine(data_home, ".local/share/vulkan/implicit_layer.d");
+    }
+
+    for (QString layer : vkLayerList) {
+        QString filePath = FS::PathCombine(layer, "MangoHud.json");
+        if (!QFile::exists(filePath))
+            continue;
+
+        auto conf = Json::requireDocument(filePath, "MangoHud");
+        return conf["layer"]["library_path"].toString();
+    }
```

Comment:
This won't work for any system that doesn't use these paths. Like Nix for example.

You could use `QStandardPaths::standardLocations(QStandardPaths::GenericDataLocation)`, which should give you paths such as `"~/.local/share", "/usr/local/share", "/usr/share"` on Linux (as per documentation).
This does not cover `/etc` though, so we could leave that one.

Reply:
Okay, I've gone through how nix handles this and looked into the code of vulkan-loader (which showed me that some of the assumptions were total bull)

This is the list it goes through in order of selection (defaults in bracket)
```
$XDG_DATA_DIRS (/usr/local/share/:/usr/share/)
$XDG_DATA_HOME  (~/.local/share)
/etc
$XDG_CONFIG_DIRS (/etc/xdg)
$XDG_CONFIG_HOME (~/.config)
```

Implementing it to follow these env vars would also solve the issue with nix, which has a wrapping system to make a bunch of wonky stuff work.
Example: https://github.com/NixOS/nixpkgs/blob/7a6a010c3a1d00f8470a5ca888f2f927f1860a19/pkgs/tools/graphics/mangohud/default.nix#L175